src/pytools/api/_api.py

```diff
@@ -516,7 +241,7 @@ def _to_collection(
 def validate_type(
     value: T,
     *,
-    expected_type: Union[Type[T], Tuple[Type, ...]],
+    expected_type: Union[Type[T], Tuple[Type[T], ...]],
```

Comment:
I think this might be unnecessary but to conform with `isinstance` method that is used in this function in terms of typing we might change `Type[T]` to `Type[Bound_T]` where `Bound_T = TypeVar("Bound_T", bound=T)` because `isinstance` will return `True` for subclasses of `T` also, according to mro.

```
class A(): pass
class B(A): pass
isinstance(B(), A) # True
```

Reply:
Unless I'm mistaken we don't need to introduce `Bound_T` because `T` already imposes an upper bound, i.e. `Type[T]` matches all subclasses of the class bound to `T`.

Reply:
Sure! It's clear to me.